odl/tomo/operators/xray_trafo.py

```diff
@@ -176,57 +183,145 @@ def _call(self, inp):
     @property
     def adjoint(self):
         """Return the adjoint operator."""
-        return self._adjoint
-
+        return XrayBackProjector(self.domain, self.geometry, self.backend,
+                                 **self.kwargs)
 
-class DiscreteXrayTransformAdjoint(Operator):
 
-    """The adjoint of the discrete X-ray transform."""
+class XrayBackProjector(Operator):
+    """The adjoint of the discrete X-ray transform between `L^p` spaces."""
 
-    def __init__(self, forward):
+    def __init__(self, discr_range, geometry, backend='astra', **kwargs):
         """Initialize a new instance.
 
         Parameters
         ----------
-        forward : `DiscreteXrayTransform`
-            An instance of the discrete X-ray transform
+        discr_range : `odl.DiscreteLp`
+            Reconstruction space, the range of the back-projector
+        geometry : `Geometry`
+            The geometry of the transform, contains information about
+            the operator domain. It needs to have a sampling grid for
+            motion and detector parameters.
+        backend : {'astra', 'astra_cuda', 'astra_cpu'}, optional
+            Implementation back-end for the transform. Supported back-ends:
+            'astra': ASTRA toolbox, uses CPU or CUDA depending on the
+            underlying data space of ``discr_range``
```

Comment:
Agree, we can add it later, also we can allow users to pick another than the one they are providing, perhaps they prefer using the CPU projector with CUDA data for some reason, then we should allow that.